### PR TITLE
KUBE-566: send telemetry to mothership

### PR DESCRIPTION
**Pull request by ValyaB**

### CODE REVIEW

File: internal/monitor/metadata.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	return nil
+}
+
+// watchForMetadataChanges starts a watch on a local file for updates and returns changes to metadata channel. watcher stops when context is done
+func watchForMetadataChanges(ctx context.Context, log logrus.FieldLogger, metadataFilePath string) (chan Metadata, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("setting up new watcher: %w", err)
+	}
+	updates := make(chan Metadata, 10)
```
Comment (aldor007) [inline internal/monitor/metadata.go:51]: Why it is a buffering channel?

File: internal/config/config.go [hunk 2]
```diff
@@ -17,8 +25,15 @@ type Config struct {
 	ClusterID      string
 	PprofPort      int
 	LeaderElection LeaderElection
-	PodName        string
-	NodeName       string
+
+	MonitorMetadata string `mapstructure:"monitor_metadata"`
```
Comment (aldor007) [inline internal/config/config.go:29]:
```suggestion
	MonitorMetadataPath string `mapstructure:"monitor_metadata"`
```

As far I can see this is path to the file but name is a bit confusing.

File: internal/monitor/monitor.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	if err != nil {
+		return fmt.Errorf("setting up metadata watch: %w", err)
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case metadata := <-metadataUpdates:
+			clusterIDHandler(metadata.ClusterID)
```
Comment (aldor007) [inline internal/monitor/monitor.go:35]: Why do we need to update clusterID base on metadata file changes. ClusterID should be passed via env on the start and shouldn't change

Comment (ValyaB) [reply internal/monitor/monitor.go:35]: Let me clean this logic, which left after the agent

**Review by stgleb — APPROVED** [review]
LGTM

File: internal/monitor/metadata.go (hunk 1 above)
Comment (ValyaB) [reply internal/monitor/metadata.go]: added non blocking write to channel and tests

**Review by aldor007 — APPROVED** [review]

**Pull request merged**